pages/items/[id]/index.js

```diff
@@ -0,0 +1,47 @@
+import useSWR from "swr";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+
+import ItemDetails from "@/components/ItemDetails";
+
+export default function DetailsPage() {
+  const router = useRouter();
+  const { isReady } = router;
+  const { id } = router.query;
+
+  const { data: item, mutate, isLoading, error } = useSWR(`/api/items/${id}`);
+
+  async function handleStatus() {
+    const newItem = { ...item, inDiscuss: !item.inDiscuss };
+    await fetch(`/api/items/${id}`, {
+      method: "PUT",
+      body: JSON.stringify(newItem),
+    });
+    mutate(newItem);
+  }
+
+  useEffect(() => {
+    if (!isReady) {
+      return <h2>Loading...</h2>;
+    }
+  }, [isReady]);
+
```

Comment:
why is this wrapped in a `useEffect`?

keep in mind that the return within a `useEffect` is the cleanup function for that effect.

Reply:
Hi Michael, 
thank you for checking! `useEffect` because of commit from Thomas:

[ThomasF85](https://github.com/ThomasF85) [yesterday](https://github.com/Olga-Mishareva/capstone-project-lost-n-found/pull/7#discussion_r1086420865)

> Sorry about my first comment. Using isLoading like this (for conditional rendering) is perfectly fine! But router.isReady should actually only be used in a useEffect and not for conditional rendering, see the [Next.JS router docs](https://nextjs.org/docs/api-reference/next/router#router-object).

I found only that Example in Internet. Should I use isReady like this?

```
useEffect(() => {
    if (!isReady) return;
  }, [isReady]);
```